# Блок 1: Теория вероятности и логика

### Задание 1: Фермер
На ферме содержатся шесть разных видов животных, и каждый раз, когда фермер заходит в сарай, он видит одно случайное животное. За день фермер заходит в сарай 6 раз.
Каково математическое ожидание количества разных видов животных, которые фермер увидит за день?

In [1]:
vid = 6
zahod = 6
answer = vid * (1 - (((vid - 1)/vid) ** zahod))
print(round(answer, 2))

3.99


В среднем фермер за день будет видеть 3.99 (~4) разных видов животных

### Задание 2: Кулинарное соревнование
В конкурсе участвуют 80 шеф-поваров с уникальными уровнями мастерства. В первом этапе судьи случайным образом распределяют их по парам (в любом состязании двух шефов выигрывает тот, у кого выше уровень мастерства). На втором этапе шефы снова случайно образуют пары для финального раунда (пары могут повториться). Победную награду получают те, кто выиграл в обоих этапах.
Каково математическое ожидание числа победителей?

In [2]:
sum = 0
for k in range(1, 81):
    sum += ((k - 1) / 79) ** 2
print(round(sum, 1))

26.8


### Задание 3: Одинокая дорога
На пустынном шоссе вероятность появления автомобиля за 30-минутный период составляет 0.95.
Какова вероятность его появления за 10 минут? А за 27 минут?

_1 - e^(-30*lambda_1) = 0.95, откуда lambda_1 = - ln(0.05)/30_

In [10]:
import numpy as np
lambda_1 = -np.log(0.05) / 30
a_10 = 1 - np.exp(-lambda_1 * 10)
a_27 = 1 - np.exp(-lambda_1 * 27)
print(round(a_10 * 100, 1), round(a_27 *100, 1), sep='; ')

63.2; 93.3
